File: examples/29-handout-master/main.go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/djinn-soul/gopptx/pkg/pptx"
	"github.com/djinn-soul/gopptx/pkg/pptx/handout"
)

const outputDir = "examples/output"

func main() {
	if err := run(); err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}

func run() error {
	if err := os.MkdirAll(outputDir, 0o750); err != nil {
		return fmt.Errorf("create output directory: %w", err)
	}

	slides := []pptx.SlideContent{
		pptx.NewSlide("Handout Master Demo").AddBullet("Demonstrates handout master support."),
		pptx.NewSlide("Slide 2").AddBullet("More content here."),
		pptx.NewSlide("Slide 3").AddBullet("Third slide."),
	}

	// --- 1. Default handout master (1-up, all placeholders visible) ---
	if err := pptx.NewPresentationBuilder("Handout Default").
		WithMetadata(pptx.Metadata{HandoutMaster: handout.New()}).
		AddSlide(slides[0]).AddSlide(slides[1]).AddSlide(slides[2]).
		WriteToFile(outputDir + "/29_handout_master_default.pptx"); err != nil {
		return fmt.Errorf("save default: %w", err)
	}
	log.Println("Saved: 29_handout_master_default.pptx")

	// --- 2. 6-up layout with custom header/footer, date hidden ---
	if err := pptx.NewPresentationBuilder("Handout 6-Up").
		WithMetadata(pptx.Metadata{
			HandoutMaster: handout.New().
				WithLayout(handout.Layout6Up).
				WithHeader("Acme Corp — Q1 Review").
				WithFooter("Confidential").
				HideDate(),
		}).
		AddSlide(slides[0]).AddSlide(slides[1]).AddSlide(slides[2]).
		WriteToFile(outputDir + "/29_handout_master_6up.pptx"); err != nil {
		return fmt.Errorf("save 6-up: %w", err)
	}
	log.Println("Saved: 29_handout_master_6up.pptx")

	// --- 3. Outline layout, all placeholders hidden ---
	if err := pptx.NewPresentationBuilder("Handout Outline").
		WithMetadata(pptx.Metadata{
			HandoutMaster: handout.New().
				WithLayout(handout.LayoutOutline).
				HideHeader().
				HideFooter().
				HideDate().
				HidePageNumber(),
		}).
		AddSlide(slides[0]).
		WriteToFile(outputDir + "/29_handout_master_outline.pptx"); err != nil {
		return fmt.Errorf("save outline: %w", err)
	}
	log.Println("Saved: 29_handout_master_outline.pptx")

	return nil
}

File: examples/29-handout-master/main.py
"""Handout master example in Python.

Configures handout layouts and writes multiple output variants.
"""

from __future__ import annotations

from pathlib import Path

from gopptx import Presentation


def build_base_slides(prs: Presentation) -> None:
    prs.update_slide(
        0, title="Handout Master Demo", bullets=["Demonstrate handout modes"]
    )
    prs.add_slide("Slide 2", bullets=["Additional content"])
    prs.add_slide("Slide 3", bullets=["Third slide content"])


def main() -> None:
    root = Path(__file__).resolve().parents[2]
    output_dir = root / "examples" / "output"
    output_dir.mkdir(parents=True, exist_ok=True)

    default_path = output_dir / "29_handout_master_default.pptx"
    six_up_path = output_dir / "29_handout_master_6up.pptx"
    outline_path = output_dir / "29_handout_master_outline.pptx"

    with Presentation.new("Handout Master Default (Python)") as prs:
        build_base_slides(prs)
        prs.save(str(default_path))

    with Presentation.new("Handout Master 6-Up (Python)") as prs:
        build_base_slides(prs)
        prs.update_handout_master(
            orientation="landscape",
            slides_per_page=6,
        )
        prs.save(str(six_up_path))
        print(f"6-up handout metadata: {prs.get_handout_master()}")

    with Presentation.new("Handout Master Outline (Python)") as prs:
        prs.update_slide(
            0, title="Outline Mode", bullets=["Single-slide outline output"]
        )
        prs.update_handout_master(
            orientation="portrait",
            slides_per_page=1,
        )
        prs.save(str(outline_path))

    print(f"Created: {default_path}")
    print(f"Created: {six_up_path}")
    print(f"Created: {outline_path}")


if __name__ == "__main__":
    main()
